### ISSUE TITLE
xcube serve Datasets configuration to use s3 resources

### ISSUE THREAD

**Issue by AliceBalfanz** [dcs4cop/xcube#828]
**Describe the bug**
When configuring Datasets via xcube serve not using the DataStore framework, s3 resources cause errors. 

**To Reproduce**
Use the configuration: 

```yaml

base_dir: s3://xcube-viewer-app/ws/dev

Viewer:
  Configuration:
    Path: viewer

Datasets:
  - Identifier: app-l2c-cyanoalert-olci-balt-v2
    Path: water-services/Sweden/app-l2c-cyanoalert-olci-balt-Sweden-v2.zarr
    Title: Sverige app v2 - limited temporal length
    FileSystem: s3
    Endpoint: "https://s3.eu-central-1.amazonaws.com"
    Anonymous: false
    Style: default

  - Identifier: c-app-l2c-cyanoalert-olci-balt-v2-1D
    Path: api/augmentation/resample_in_time.py
    Title: Sverige app v2 1D- limited temporal length
    Style: default
    FileSystem: memory
    Function: compute_dataset
    InputDatasets: [app-l2c-cyanoalert-olci-balt-v2]
    InputParameters:
      period: 1D
```

Traceback:
```
[I 2023-03-03 12:06:50,791 xcube] Opened dataset 'app-l2c-cyanoalert-olci-balt-v2' from data store 's3_1': took 326.93ms
[W 2023-03-03 12:06:50,792 xcube] Skipping dataset app-l2c-cyanoalert-olci-balt-v2: Parameter validation failed:
Invalid bucket name "s3:": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/controllers.py", line 140, in get_datasets
    get_dataset(ctx, ds_id,
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/controllers.py", line 182, in get_dataset
    ml_ds = ctx.get_ml_dataset(ds_id)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 183, in get_ml_dataset
    ml_dataset, _ = self._get_dataset_entry(ds_id)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 602, in _get_dataset_entry
    self._set_dataset_entry(self._create_dataset_entry(ds_id))
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 615, in _create_dataset_entry
    ml_dataset = self._open_ml_dataset(dataset_config)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 639, in _open_ml_dataset
    dataset = data_store.open_data(data_id, **open_params)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/core/store/fs/store.py", line 333, in open_data
    return opener.open_data(fs_path,
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/core/store/fs/impl/dataset.py", line 190, in open_data
    fs.exists(f'{data_id}/.zmetadata'))
  File "/opt/conda/lib/python3.9/site-packages/fsspec/asyn.py", line 114, in wrapper
    return sync(self.loop, func, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/fsspec/asyn.py", line 99, in sync
    raise return_result
  File "/opt/conda/lib/python3.9/site-packages/fsspec/asyn.py", line 54, in _runner
    result[0] = await coro
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 974, in _exists
    await self._info(path, bucket, key, version_id=version_id)
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 1238, in _info
    out = await self._call_s3(
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 339, in _call_s3
    return await _error_wrapper(
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 139, in _error_wrapper
    raise err
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 112, in _error_wrapper
    return await func(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/client.py", line 321, in _make_api_call
    request_dict = await self._convert_to_request_dict(
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/client.py", line 383, in _convert_to_request_dict
    api_params = await self._emit_api_params(
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/client.py", line 416, in _emit_api_params
    await self.meta.events.emit(
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/hooks.py", line 66, in _emit
    response = await resolve_awaitable(handler(**kwargs))
  File "/opt/conda/lib/python3.9/site-packages/botocore/handlers.py", line 275, in validate_bucket_name
    raise ParamValidationError(report=error_msg)
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid bucket name "s3:": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
```

**Expected behavior**
I would expect that the Datasets will be able to use the s3 resources the same way the DataStore configuration is.

**Comment by forman** [comment]
Please try commenting out the `Endpoint` configuration.

**Comment by AliceBalfanz** [comment]
When I comment out `Endpoint`, the following error appears: 

```
[I 2023-03-06 16:18:24,016 xcube] Opened dataset 'app-l2c-cyanoalert-olci-balt-v2' from data store 's3_1': took 854.25ms
[I 2023-03-06 16:18:24,624 xcube] Loading details for dataset c-app-l2c-cyanoalert-olci-balt-v2-1D
/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/context.py:849: UserWarning: chunk cache size is not effective for datasets computed from scripts
  warnings.warn(
[W 2023-03-06 16:18:24,630 xcube] Python sys.path prepended by /tmp/xcube-byoa1hhcs1qm
[I 2023-03-06 16:18:24,631 xcube] Opened memory dataset /home/alicja/Desktop/projects/s3:/xcube-viewer-app/ws/api/augmentation/resample_in_time.py: took 2.01ms
[I 2023-03-06 16:18:24,631 xcube] Opened dataset 'c-app-l2c-cyanoalert-olci-balt-v2-1D' from 'memory': took 6.07ms
[W 2023-03-06 16:18:24,631 xcube] Skipping dataset c-app-l2c-cyanoalert-olci-balt-v2-1D: HTTP status 580: Invalid dataset descriptor 'c-app-l2c-cyanoalert-olci-balt-v2-1D': No module named 'resample_in_time'
Traceback (most recent call last):
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/mldataset/computed.py", line 129, in get_callable
    callable_obj = CodeConfig.from_file_set(
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/byoa/config.py", line 294, in get_callable
    self.set_callable(self._load_callable())
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/byoa/config.py", line 314, in _load_callable
    return _load_callable(code_config.file_set.path,
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/byoa/config.py", line 435, in _load_callable
    module = importlib.import_module(module_name)
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'resample_in_time'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/controllers.py", line 140, in get_datasets
    get_dataset(ctx, ds_id,
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/controllers.py", line 182, in get_dataset
    ml_ds = ctx.get_ml_dataset(ds_id)
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/context.py", line 181, in get_ml_dataset
    ml_dataset, _ = self._get_dataset_entry(ds_id)
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/context.py", line 602, in _get_dataset_entry
    self._set_dataset_entry(self._create_dataset_entry(ds_id))
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/context.py", line 615, in _create_dataset_entry
    ml_dataset = self._open_ml_dataset(dataset_config)
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/context.py", line 655, in _open_ml_dataset
    ml_dataset = _open_ml_dataset_from_python_code(
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/webapi/datasets/context.py", line 861, in _open_ml_dataset_from_python_code
    return open_ml_dataset_from_python_code(
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/mldataset/computed.py", line 225, in open_ml_dataset_from_python_code
    return _open_ml_dataset_from_python_code(
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/mldataset/computed.py", line 275, in _open_ml_dataset_from_python_code
    return ComputedMultiLevelDataset(
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/mldataset/computed.py", line 59, in __init__
    callable_ref, callable_obj = self.get_callable(
  File "/home/alicja/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube/core/mldataset/computed.py", line 135, in get_callable
    raise exception_type(
xcube.server.api._InvalidServerConfig: HTTP status 580: Invalid dataset descriptor 'c-app-l2c-cyanoalert-olci-balt-v2-1D': No module named 'resample_in_time'
```

**Comment by forman** [comment]
Note, this issue occurs for a development version 1.0.0.dev of xcube. Will be fixed for the xcube 1.0 release.

**Comment by forman** [comment]
> When I comment out `Endpoint`, the following error appears:
> 
> ```
> [W 2023-03-06 17:14:38,386 xcube] Skipping dataset app-l2c-cyanoalert-olci-balt-v2: Parameter validation failed:
> ...
> ModuleNotFoundError: No module named 'resample_in_time'
> 
> The above exception was the direct cause of the following exception:
> ...
> xcube.server.api._InvalidServerConfig: HTTP status 580: Invalid dataset descriptor 'c-app-l2c-cyanoalert-olci-balt-v2-1D': No module named 'resample_in_time'
> ```

The issue where a Python module cannot be found after downloading from S3 is back.

`fsspec.AbstractFileSystem.get(source, target)` is completely unpredictable if `source` and `target` are directories. Presumably the problem is caused by special `s3fs.S3FileSystem` implementation. 

We get different outputs if `target` exists or not and whether `source` and/or `target` paths end with a slash or whether they are absolute paths or not! In most cases the `target` directory will contain another directory named by the last path element of `source`. But we were not able do predict the outcome.

That's why xcube cannot find the augmentation module.
